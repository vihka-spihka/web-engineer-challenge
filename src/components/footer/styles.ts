import styled from 'styled-components';

export const StyledFooter = styled.footer`
	display: flex;
	justify-content: flex-end;
	border-top: 1px solid ${props => props.theme.global.primary_border_color};
	background: white;
`;
