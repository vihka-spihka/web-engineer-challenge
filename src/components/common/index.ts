import { Input } from './input';
import { StyledButton } from './styles';
import { Loader } from './loader';

export { Input, StyledButton , Loader};
