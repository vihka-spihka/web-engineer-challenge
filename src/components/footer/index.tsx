import React from 'react';
import { StyledFooter } from './styles';

export const Footer: React.FC = () => {
	return (
		<StyledFooter>
			<img src='koala-logo.png' height='35' alt='Company logo' />
		</StyledFooter>
	);
};
