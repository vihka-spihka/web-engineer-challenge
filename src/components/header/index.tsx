import React from 'react';
import { StyledHeader } from './styles';

export const Header: React.FC = () => {
	return (
		<StyledHeader>
			<div>&#8987;</div>
			<h1>Countdown Clock</h1>
		</StyledHeader>
	);
};
