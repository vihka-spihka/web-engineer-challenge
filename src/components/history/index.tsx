import React from 'react';
import { StyledHistory } from './styles';
import { HistoryItem } from './history-item';

interface IProps {
	items: Array<Countdown>;
};

export const History: React.FC<IProps> = ({ items }) => {
	return (
		<StyledHistory>
			<h2>History</h2>
			<div>
				{items.map(
					it => <HistoryItem key={it.id} start={it.start} duration={it.duration} />
				)}
			</div>
		</StyledHistory>
	);
};
