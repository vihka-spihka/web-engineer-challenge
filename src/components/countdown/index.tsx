import React, { useState } from 'react';
import { StyledCountdown } from './styles';
import { Input, Button } from './../common';
import { COUNTDOWN_MODE } from './../../App';

interface IProp {
	mode: COUNTDOWN_MODE;
	timeLeft?: number;
	changeMode: (mode: COUNTDOWN_MODE, time?: number) => void;
};

export const Countdown = ({ mode, timeLeft, changeMode }: IProp) => {
	const [value, setValue] = useState<InputValue>('');

	const submitForm = (e: React.SyntheticEvent): void => {
		e.preventDefault();
		changeMode(COUNTDOWN_MODE.IN_PROGRESS, value as number);
		setValue('');
	};

	const editCountdown = () => {
		timeLeft && setValue(timeLeft);
		changeMode(COUNTDOWN_MODE.EDIT);
	};

	const resetCountdown = (): void => {
		setValue('');
		changeMode(COUNTDOWN_MODE.NEW);
	};

	let countdown: React.ReactElement;

	switch(mode) {
		case COUNTDOWN_MODE.EDIT:
			countdown = (
				<form onSubmit={(e: React.SyntheticEvent) => submitForm(e)}>
					<Input
						type={'number'}
						value={value?.toString()}
						label={'Enter time in seconds: '}
						onChangeInput={val => setValue(val)}
						placeholder={'e.g. 1000'} />
					<Button label={'Done'} onClick={(e: React.SyntheticEvent) => submitForm(e)} />
					<Button label={'Reset'} onClick={() => resetCountdown()} />
				</form>
			);
			break;
		case COUNTDOWN_MODE.IN_PROGRESS:
			countdown = (
				<>
					<div>Seconds left: {timeLeft}</div>
					<Button label={'Edit'} onClick={() => editCountdown()} />
					<Button label={'Reset'} onClick={() => resetCountdown()} />
				</>
			);
			break;
		case COUNTDOWN_MODE.NEW:
		default:
			countdown = (
				<form onSubmit={(e) => submitForm(e)}>
					<Input
						type={'number'}
						value={value?.toString()}
						label={'Enter time in seconds: '}
						onChangeInput={val => setValue(val)}
						placeholder={'e.g. 1000'} />
					<Button label={'Start'} onClick={(e: React.SyntheticEvent) => submitForm(e)} />
				</form>
			);
			break;
	}

	return (
		<StyledCountdown>
			{countdown}
		</StyledCountdown>
	);
};
