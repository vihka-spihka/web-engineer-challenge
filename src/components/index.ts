import { Input, StyledButton, Loader } from './common';
import { Countdown } from './countdown';
import { Footer } from './footer';
import { Header } from './header';
import { History } from './history';

export { Input, StyledButton, Loader, Countdown, Footer, Header, History };
