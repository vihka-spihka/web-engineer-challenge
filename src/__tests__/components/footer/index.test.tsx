import { shallow } from 'enzyme';
import { Footer } from './../../../components/footer';

describe('Footer', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Footer />);
		expect(wrapper).toMatchSnapshot();
	});
});
