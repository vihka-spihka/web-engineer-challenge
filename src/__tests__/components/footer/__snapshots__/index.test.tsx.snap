// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<styled.footer>
  <img
    alt="Company logo"
    height="35"
    src="koala-logo.png"
  />
</styled.footer>
`;
